gravatar/src/test/java/com/gravatar/services/ProfileServiceTests.kt

```diff
@@ -0,0 +1,121 @@
+package com.gravatar.services
+
+import com.gravatar.GravatarSdkContainerRule
+import com.gravatar.api.models.Profile
+import com.gravatar.types.Email
+import com.gravatar.types.Hash
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.mockk
+import junit.framework.TestCase.assertTrue
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import retrofit2.Response
+
+@RunWith(RobolectricTestRunner::class)
+class ProfileServiceTests {
+    @get:Rule
+    var containerRule = GravatarSdkContainerRule()
+
+    private lateinit var profileService: ProfileService
+
+    @Before
+    fun setUp() {
+        profileService = ProfileService()
+    }
+
+    @Test
+    fun `given an username when loading its profile and data is returned then result is successful`() = runTest {
+        val username = "username"
+        val mockResponse = mockk<Response<Profile>> {
+            every { isSuccessful } returns true
+            every { body() } returns mockk()
+        }
+        coEvery { containerRule.gravatarApiServiceMock.getProfileById(username, any()) } returns mockResponse
+
+        val loadProfileResponse = profileService.fetchByUsername(username)
+
+        coVerify(exactly = 1) { containerRule.gravatarApiServiceMock.getProfileById(username) }
+        assertTrue(loadProfileResponse is Result.Success)
+    }
+
+    @Test
+    fun `given an username when loading its profile but date is NOT returned then result is UNKNOWN failure`() =
+        runTest {
+            val username = "username"
+            val mockResponse = mockk<Response<Profile>> {
+                every { isSuccessful } returns true
+                every { body() } returns null
+            }
+            coEvery { containerRule.gravatarApiServiceMock.getProfileById(username, any()) } returns mockResponse
+
+            val loadProfileResponse = profileService.fetchByUsername(username)
+
+            coVerify(exactly = 1) { containerRule.gravatarApiServiceMock.getProfileById(username) }
+            assertTrue((loadProfileResponse as Result.Failure).error == ErrorType.UNKNOWN)
```

Comment:
Not sure if we should test for the other error types like `TIMEOUT` similar to the test cases in `AvatarServiceTest`

Reply:
I'll probably refactor that in another PR.